### PR TITLE
Fix a rare race condition in the UI test for layout discovery

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 1]
```diff
@@ -159,7 +159,10 @@ private void subscribeToSelectedGitRepositoryChange() {
   }
 
   @UIEffect
-  private void refreshModel(GitRepository gitRepository, List<String> skippedBranchNames) {
+  private void refreshModel(
+      GitRepository gitRepository,
+      List<String> skippedBranchNames,
+      @UI Runnable doOnUIThreadWhenReady) {
```
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:165]: This _callback hell_ is getting unmaintainable, lemme open an issue...

Comment (PawelLipski) [reply frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:165]: https://github.com/VirtusLab/git-machete-intellij-plugin/issues/620

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 2]
```diff
@@ -192,7 +196,8 @@ private void refreshModel(GitRepository gitRepository, List<String> skippedBranc
 
     if (!isMacheteFilePresent) {
       LOG.info("Machete file (${macheteFilePath}) is absent, so auto discover is running");
-      doAutomaticDiscover(macheteFilePath);
+      queueAutomaticDiscover(macheteFilePath, doOnUIThreadWhenReady);
```
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:199]: Sometimes (although veeery rarely) the model returned in `refreshGraphTableModel` method in `com.virtuslab.gitmachete.uitest.UITestSuite#discoverBranchLayout` (line 96: `int branchRowsCount = callJs("project.refreshGraphTableModel().getRowCount()");`) was the one from **before** the discovery running in the background was completed (and thus this model had zero rows, not seven).

File: uiTests/src/test/resources/common.rhino.js [hunk 3]
```diff
@@ -18,5 +18,5 @@ function getNonPublicMethod(clazz, methodName) {
 
 // Do not run on the UI thread.
 function sleep() {
-  Thread.sleep(500);
+  Thread.sleep(100);
```
Comment (PawelLipski) [inline uiTests/src/test/resources/common.rhino.js:21]: This should make spotting errors like that easier in UI tests...

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java (hunk 2 above)
Comment (micpiotrowski) [reply frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:199]: Maybe leave short comment about this fact in code

**Review by micpiotrowski — APPROVED** [review]

**Review by mkondratek — APPROVED** [review]

**Pull request merged**